src/ansys/dpf/composites/layup_info/_layup_info.py

```diff
@@ -200,6 +201,37 @@ def get_dpf_material_id_by_analyis_ply_map(
     data_source_or_streams_provider:
         DPF data source with rst file or streams_provider. The streams provider is
         available from :attr:`.CompositeModel.core_model` (under metadata.streams_provider).
+
+    Note
+    ----
+    Cache the output because the computation can be performance critical.
```

Comment:
```suggestion
    Cache the output because the computation can be performance-critical.
```
Hyphenate usage here as it is a compound adjective.